group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetMetadataManager.java

```diff
@@ -918,7 +920,7 @@ public void replay(
                     groupId,
                     topic,
                     partition,
-                    OffsetAndMetadata.fromRecord(value)
+                    OffsetAndMetadata.fromRecord(offset, value)
```

Comment:
it is interesting that the offset passed in here is new (probably not the offset that OffsetAndMetadata is referring to)

Reply:
yeah, this offset is the offset of the record, it is not the committed offset in the record.

Reply:
I wonder if that is easy to get confused. And if there is a way to make it easier to known what the parameters mean

Reply:
I agree that it is a bit confusing. Let me rename `offset` to `recordOffset` to make it clearer.